app/models/concerns/cable_ready/updatable.rb

```diff
@@ -6,6 +6,26 @@ module CableReady
   module Updatable
     extend ::ActiveSupport::Concern
 
+    class MemoryDebounceAdapter
+      include Singleton
+      include MonitorMixin
+
+      delegate_missing_to :@dict
+
+      def initialize
+        super
+        @dict = {}
+      end
+
+      def []=(key, value)
+        synchronize do
+          @dict[key] = value
+        end
+      end
```

Comment:
Should we synchronize reads also? I dont' think we're quite ready to pull in a new dependency but we should at least keep an eye out for options like [`Concurrent::Hash`](https://ruby-concurrency.github.io/concurrent-ruby/master/Concurrent/Hash.html) as we chart our way into this territory.

Reply:
yep... I was looking at https://github.com/hopsoft/coin/blob/master/lib/coin/vault.rb, and since you don't synchronize reads there it confirmed my gut feeling that it shouldn't be necessary. But I'll happily add it if you think it's necessary.